### PR TITLE
Refactor Mail class: move to the top level to be used with both sandbox and sending APIs

### PR DESCRIPTION
**Pull request by r4do**
## Motivation
- Mailtrap::Sending::Base isn't base class for Mailtrap::Sending, rather for Mailtrap::Sending::Mail
- Sandbox and sending APIs send methods should have same interface, thus using same `Mail` objects, including `from_message`

## Changes
- Moved `Mail` classes upstairs.

### CODE REVIEW

File: spec/fixtures/vcr_cassettes/Mailtrap_ActionMailer_DeliveryMethod/_deliver_/converts_the_message_and_sends_via_API.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ http_interactions:
...
...railsware.com"},{"email":"bcc_2@railsware.com"}],"subject":"You are awesome!","text":"Some text","html":"<div>HTML part</div>","attachments":[{"content":"VGhpcyBpcyBhIHRleHQgZmlsZQo=","type":"text/plain","filename":"file.txt","disposition":"attachment","content_id":"txt_content_id@test.mail"},{"content":"iVBORw0KGgoAAAANSUhEUgAAABgAAAAYCAYAAADgdz34AAAABmJLR0QA/wD/AP+gvaeTAAABiElEQVRIid3VSW5UMRCA4U8d0gIx7AjhAghBTgFICAkxRFHukDDDIRiuwI7xPNAMISQRV4CABFmkWbie2nrt9muaFZTkxavhr7Jfucz/Ln2s4hU28B0/sBO6VczPCl/GNoYd6wsuZ3ELWKuBe3iSAQa4g7M4jEM4jRt4g5+4lMEHEbc+KUED/xVOvY5iThXgg/gek+UMfq4CbstU8L7RmU/c3qxwUkc0TnN/CT9Ycn4djrenhB/H24j5iMXQX8TTUsCncD6T6daUtzyp8hNSV30uJfgWAUfje70AqMFF7FC6kGOy20rQPoKTBd1ii3EsbF9LCUpH1K62q1uWwr5RSvAyjHdb+rzqSZU38iB8npWMTZu+M96mzU5qfT6H98FYKTn0sRUONwv2hQqcNK+G2FSZsNeNRsX5CqwtF7CHfVzpcn6cJbmlfqsPSJXvRczDaarpZUmaf3JP6pAjsZZw3+jM9/FIffKOyTXpRnY9OJu4+ifgXOaljnghtedurA94HraZn8x/Q34DYaON8Fk9Z1IAAAAASUVORK5CYII=","type":"image/png","filename":"file.png","disposition":"inline","content_id":"png_content_id@test.mail"}],"headers":{"Reply-To":"reply-to@railsware.com","X-Special-Domain-Specific-Header":"SecretValue","One-more-custom-header":"CustomValue"},"category":"Module Test"}'
```
Comment (r4do) [inline spec/fixtures/vcr_cassettes/Mailtrap_ActionMailer_DeliveryMethod/_deliver_/converts_the_message_and_sends_via_API.yml:8]: for all fixtures change is the same - updated order of variables, we now have `from` first, `subject` and content fields upstairs.

**Review by ayanko — COMMENTED** [review]
IMHO CHANGELOG should mention about this breaking changes...

**Comment by r4do** [comment]
I'm going to create a release in next PR, so will update changelog there (it mentions releases only).

**Review by leonid-shevtsov — APPROVED** [review]
Cool! Now i'm thinking, do we even need a special mail object, or could we possibly reuse existing and well-known Mail class from the `mail` gem. But it's not for right now.

**Comment by r4do** [comment]
> could we possibly reuse existing and well-known Mail class from the `mail` gem.

I think we can experiment, but in this case we'll need to extend it with our own `as_json` method.

**Pull request merged**